src/Tools/dotnet-monitor/DiagnosticsMonitorCommandHandler.cs

```diff
@@ -68,13 +70,33 @@ public async Task<int> Start(CancellationToken token, IConsole console, string[]
             using IHost host = CreateHostBuilder(console, urls, metricUrls, metrics, diagnosticPort, noAuth).Build();
             try
             {
-                await host.RunAsync(token);
+                await host.StartAsync(token);
+
+                await host.WaitForShutdownAsync(token);
             }
             catch (MonitoringException)
             {
                 // It is the responsibility of throwers to ensure that the exceptions are logged.
                 return -1;
             }
+            catch (OptionsValidationException ex)
+            {
+                host.Services.GetRequiredService<ILoggerFactory>()
+                    .CreateLogger(typeof(DiagnosticsMonitorCommandHandler))
+                    .OptionsValidationFailure(ex);
+                return -1;
+            }
+            finally
+            {
+                if (host is IAsyncDisposable asyncDisposable)
```

Comment:
Does C#'s `await using (...)` do this same logic and accept an `IDisposable` or `IAsyncDisposable`, or does it require an `IAsyncDisposable`?

Reply:
I'll give that a try, but I believe that `await using` requires it to implement IAsyncDisposable.

Reply:
Results in compilation error CS8417.